modules/infra/api/graph.c

```diff
@@ -44,8 +48,27 @@ static struct api_out graph_dump(const void * /*request*/, void **response) {
 		goto end;
 	}
 
+	if (!(req->flags & GR_INFRA_GRAPH_DUMP_F_ERRORS)) {
+		char *copy = resp->dot;
+		char *line = buf;
+		char *eol;
+
+		while ((eol = strchr(line, '\n')) != NULL) {
+			*eol = '\0';
+			// Remove sink nodes from the output. They all have a "darkorange" color.
+			// Also remove non-sink nodes that contain "error" in their name.
+			if (!strstr(line, "darkorange") && !strstr(line, "error")) {
+				*eol = '\n'; // restore newline char
+				copy = memccpy(copy, line, 0, eol - line + 1);
```

Comment:
Should be memccpy(.., '\n', ..)

Reply:
you think I would have tested again ? :)